fdbserver/LogSystemDiskQueueAdapter.actor.cpp

```diff
@@ -42,19 +42,19 @@ class LogSystemDiskQueueAdapterImpl {
 							break;
 						}
 						when( wait( self->localityChanged ) ) {
-							self->cursor = self->logSystem->peekSpecial( UID(), self->recoveryLoc, self->tag, self->peekLocality ? self->peekLocality->get().primaryLocality : tagLocalityInvalid, self->peekLocality ? self->peekLocality->get().knownCommittedVersion : invalidVersion );
+							self->cursor = self->logSystem->peekTxs( UID(), self->recoveryLoc, self->peekLocality ? self->peekLocality->get().primaryLocality : tagLocalityInvalid, self->peekLocality ? self->peekLocality->get().knownCommittedVersion : invalidVersion );
 							self->localityChanged = self->peekLocality->onChange();
 						}
 						when( wait( delay(self->peekTypeSwitches==0 ? SERVER_KNOBS->DISK_QUEUE_ADAPTER_MIN_SWITCH_TIME : SERVER_KNOBS->DISK_QUEUE_ADAPTER_MAX_SWITCH_TIME)) ) {
 							self->peekTypeSwitches++;
 							if(self->peekTypeSwitches%3==1) {
-								self->cursor = self->logSystem->peek( UID(), self->recoveryLoc, self->tag, true );
+								self->cursor = self->logSystem->peekTxs( UID(), self->recoveryLoc, tagLocalityInvalid, invalidVersion );
 								self->localityChanged = Never();
 							} else if(self->peekTypeSwitches%3==2) {
-								self->cursor = self->logSystem->peekSpecial( UID(), self->recoveryLoc, self->tag, self->peekLocality ? self->peekLocality->get().secondaryLocality : tagLocalityInvalid, self->peekLocality ? self->peekLocality->get().knownCommittedVersion : invalidVersion );
+								self->cursor = self->logSystem->peekTxs( UID(), self->recoveryLoc, self->peekLocality ? self->peekLocality->get().secondaryLocality : tagLocalityInvalid, self->peekLocality ? self->peekLocality->get().knownCommittedVersion : invalidVersion );
 								self->localityChanged = self->peekLocality->onChange();
 							} else {
-								self->cursor = self->logSystem->peekSpecial( UID(), self->recoveryLoc, self->tag, self->peekLocality ? self->peekLocality->get().primaryLocality : tagLocalityInvalid, self->peekLocality ? self->peekLocality->get().knownCommittedVersion : invalidVersion );
+								self->cursor = self->logSystem->peekTxs( UID(), self->recoveryLoc, self->peekLocality ? self->peekLocality->get().primaryLocality : tagLocalityInvalid, self->peekLocality ? self->peekLocality->get().knownCommittedVersion : invalidVersion );
 								self->localityChanged = self->peekLocality->onChange();
 							}
 						}
```

Comment:
I had to put a `wait(delay(0))` above the `continue;` on line 69, because `getMore` returns when there's a message, or when the version changes, and it appears we now have a very large number of version changes in a row.  When that happens, there's a loop of wait(getMore) -> break -> !hasMessage() -> continue -> wait(getMore) that overflows the stack.